h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java

```diff
@@ -613,4 +613,29 @@ private final int getXGBoostClassIndex(final String treeClass) {
     return treeClassIndex;
   }
 
+  //--------------------------------------------------------------------------------------------------------------------
+  // Serialization into a POJO
+  //--------------------------------------------------------------------------------------------------------------------
+
+  @Override
+  protected boolean toJavaCheckTooBig() {
+    return _output == null || _output._ntrees * _parms._max_depth > 1000;
```

Comment:
The line drawn here is arbitrary ?

Reply:
yes pretty much - something like what constitutes an amount of code a human would go through